kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/GenerateIcebergCompatActionUtils.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.actions;
+
+import static io.delta.kernel.internal.data.TransactionStateRow.*;
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+import io.delta.kernel.Transaction;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.exceptions.KernelException;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.internal.TableConfig;
+import io.delta.kernel.internal.data.TransactionStateRow;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.icebergcompat.IcebergCompatV2MetadataValidatorAndUpdater;
+import io.delta.kernel.utils.CloseableIterable;
+import io.delta.kernel.utils.DataFileStatus;
+import java.net.URI;
+import java.util.Map;
+
+/** Utilities to convert Iceberg add/removes to Delta Kernel add/removes */
+public final class GenerateIcebergCompatActionUtils {
+
+  /**
+   * Create an add action {@link Row} that can be passed to {@link Transaction#commit(Engine,
+   * CloseableIterable)} from an Iceberg add.
+   *
+   * @param transactionState the transaction state from the built transaction
+   * @param fileStatus the file status to create the add with (contains path, time, size, and stats)
+   * @param partitionValues the partition values for the add
+   * @param dataChange whether or not the add constitutes a dataChange (i.e. append vs. compaction)
+   * @return add action row that can be committed to the transaction
```

Comment:
nit (verbiage): we _include_ actions in transactions, and _commit_ transactions to tables, right?

Reply:
Idk I feel like we also commit actions in transactions :) but happy to change this